proxy/proxy.go

```diff
@@ -333,17 +333,17 @@ func (proxy *proxy) init() error {
 	return nil
 }
 
-var nextClientID = uint64(1)
+var nextClientID = uint64(0)
 
 func (proxy *proxy) serveNewClient(proto *protocol, newConn net.Conn) {
+	atomic.AddUint64(&nextClientID, 1)
```

Comment:
I think this line deserves a comment explaining why the increment is happening here.